__tests__/app.test.js

```diff
@@ -84,54 +105,125 @@ describe("GET /api/articles/:article_id", () => {
 });
 
 describe("GET /api/articles", () => {
-  test("200: responds with an array of articles containing the correct properties", () => {
+  test("200: filters articles by topic", () => {
     return request(app)
-      .get("/api/articles")
+      .get("/api/articles?topic=technology")
       .expect(200)
       .then(({ body }) => {
-        expect(Array.isArray(body.articles)).toBe(true);
-        expect(body.articles).toHaveLength(13);
+        expect(body.articles).toBeInstanceOf(Array);
         body.articles.forEach((article) => {
-          expect(article).toEqual(
-            expect.objectContaining({
-              article_id: expect.any(Number),
-              author: expect.any(String),
-              title: expect.any(String),
-              topic: expect.any(String),
-              created_at: expect.any(String),
-              votes: expect.any(Number),
-              article_img_url: expect.any(String),
-              comment_count: expect.any(Number),
-            })
-          );
+          expect(article.topic).toBe("technology");
         });
       });
   });
-  test("200: responds with articles sorted by created_at in descending order", () => {
+
+  test("200: returns an empty array if the topic does not exist", () => {
+    return request(app)
+      .get("/api/articles?topic=nonexistent")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toEqual([]);
```

Comment:
This would actually be a 404 if the topic doesn't exist.
You are correct though in that we should be returning a 200 with an empty array, but this would be when it's a valid topic that has no associated articles (I believe "paper" is the one in the test database!) 🙂